### PR TITLE
Add documentation to DG for `undo`, `redo` and `history`

### PR DESCRIPTION
**Pull request by vigneshsankariyer1234567890**
Details the implementation of the various classes which allow the history of states to be stored.

Here are some of the main diagrams which are being used:
HistoryCommandSequenceDiagram
HistorySequenceToStoreCommandsDiagram
ModelClassDiagram
UndoSequenceDiagram
DownButtonActivityDiagram
UpButtonActivityDiagram

### CODE REVIEW

File: docs/DeveloperGuide.md
Comment (skythefire) [inline docs/DeveloperGuide.md:446]: I think there is a typo here!

File: docs/DeveloperGuide.md
Comment (skythefire) [inline docs/DeveloperGuide.md:456]: Same as above!

**Review by skythefire — APPROVED** [review]
LGTM, after fixing the typos (if they are indeed typos)!

File: docs/DeveloperGuide.md
Comment (tanqiya) [reply docs/DeveloperGuide.md:446]: it's intentional. referencing object name in the diagram

Comment (skythefire) [reply docs/DeveloperGuide.md:446]: I see

**Comment by skythefire** [comment]
For the Model Class Diagram, the multiplicity of Person to the optional fields ie. Email, Address, and Remark are 1 because in our implementation we give them default values?

**Comment by vigneshsankariyer1234567890** [comment]
> For the Model Class Diagram, the multiplicity of Person to the optional fields ie. Email, Address, and Remark are 1 because in our implementation we give them default values?

i wasnt thinking along those lines per se, more of the fact that Person and it's fields are immutable. This should allow the multiplicity to be 1

**Comment by gok99** [comment]
image

Should this be `c:HistoryCommand`?

**Comment by gok99** [comment]
Activity diagram conditions should be in box brackets (`[]`)

image

**Comment by vigneshsankariyer1234567890** [comment]
> image
> 
> Should this be `c:HistoryCommand`?
> 
> image
> 
> Perhaps could use `c:XYZCommand` with a note on what `XYZCommand` is, like in the Logic diagram:
> 
> image

good catch bro, i'll fix it asap before the next merge.

**Comment by vigneshsankariyer1234567890** [comment]
> Activity diagram conditions should be in box brackets (`[]`)
> 
> image

will rectify this as well

**Pull request merged**
